src/charm.py

```diff
@@ -124,6 +124,24 @@ def is_main(self) -> bool:
         """
         return self.get_main_unit() == self.unit.name
 
+    def get_unit_number(self, unit_name: str = "") -> str:
+        """Get unit number from unit name.
+
+        Args:
+            unit_name: unit name. E.g.: synapse/0
+
+        Returns:
+            Unit number. E.g.: 0
+        """
+        if not unit_name:
+            unit_name = self.unit.name
+        match = re.search(r"[-\/](\d+)", unit_name)
```

Comment:
The regex also matches `-0` in `synapse-0` (I think you replace / with - in another part of the code). You might want to add this to the docstring (otherwise it might be a bit confusing for the reader of the code).

Reply:
Indeed, but I'm using the first group, not the match :)

This is a nice tool that I usually test regular expressions:
https://regex101.com/r/YHoPN0/1

I'll add it to docstring.

Reply:
Actually...From your comment and @yhaliaw I'll remove the regex and replace for something like this. Is more readable?

```python
    index = unit_name.rfind('/') # synapse/0 pattern
    if index == -1:
        index = unit_name.rfind('-') # synapse-0.synapse-endpoints pattern
   return unit_name[index + 1:]
```

Reply:
Replaced by split + rfind :)